vspace/src/main/java/edu/asu/diging/vspace/web/HomeController.java

```diff
@@ -20,13 +22,18 @@
 
     @RequestMapping(value = "/")
     public String home(Model model) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
         if (!setupManager.isSetup()) {
             return "setup";
         }
 
         IExhibition exhibition = exhibitionManager.getStartExhibition();
         if (exhibition != null && exhibition.getStartSpace() != null) {
-            return "redirect:/exhibit/space/" + exhibition.getStartSpace().getId();
+            //
+            if(authentication.getPrincipal()!="anonymousUser")
```

Comment:
- in java strings are compared with `equals` (e.g. `string1.equals(string2)`)
- the username for anonymous users can be configured I believe, so this is error prone; it could be changed and then this will break; it's probably safer to check if the authentication object is of type `AnonymousAuthenticationToken`.